### PR TITLE
expose IssueLabel.description; update labels REST APIs

### PR DESCRIPTION
**Pull request by devoncarew**
- expose the IssueLabel `description` field
- pass a `description` param to `createLabel`
- rename `editLabel` to `updateLabel` to match the github API; update the implementation to use PATCH, and update the available params (`new_name`, `color`, `description`) to match the API

### CODE REVIEW

File: lib/src/common/issues_service.dart [hunk 1]
```diff
@@ -290,19 +290,41 @@ class IssuesService extends Service {
...
 
-  /// Edits a label.
+  /// Update a label.
   ///
   /// API docs: https://developer.github.com/v3/issues/labels/#update-a-label
-  Future<IssueLabel> editLabel(RepositorySlug slug, String name, String color) {
-    return github.postJSON('/repos/${slug.fullName}/labels/$name',
-        body: GitHubJson.encode({'name': name, 'color': color}),
-        convert: IssueLabel.fromJson);
+  Future<IssueLabel> updateLabel(
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply lib/src/common/issues_service.dart:312]: Sure, happy to.

I did consider the breaking change thing, but given the current impl. of that method I'd guess that nobody's using it (happy to adding a forwarding method in any case).

**Comment by devoncarew** [comment]
(ptal, updated)

**Comment by devoncarew** [comment]
@robrbecker - can you take another look at this PR? We have a tool which normalizes the labels used across the `dart-lang` repos (https://github.com/dart-lang/ecosystem/tree/main/pkgs/repo_manage), but it does require this fix:) I believe that the PR as written is a 'new api' change (`9.10.1` ==> `9.11.x`), rather than a major version bump.

**Review by robrbecker — APPROVED** [review]

**Comment by robbecker-wf** [comment]
Released in version 9.8.0 https://github.com/SpinlockLabs/github.dart/releases/tag/9.8.0

**Pull request merged**